Interop/Indexer/src/main/kotlin/org/jetbrains/kotlin/native/interop/indexer/J2ObjCParser.kt

```diff
@@ -104,6 +103,26 @@ class J2ObjCParser: ClassVisitor(Opcodes.ASM5) {
     }
   }
 
+  /**
+   * Parses the methods in the class to find getters/setters and generates properties accordingly
+   *
+   * @param methods List of ObjCMethods in the class
+   * @return List of ObjCProperty based off of the getters/setters in class
+   */
+  private fun parseForProperties(methods: List<ObjCMethod>): List<ObjCProperty> {
+    val getters = methods.filter{(it.selector.startsWith("get")) && it.parameters.isEmpty()}
+
+    return getters.map{
+      val varName = it.selector.substring(3)
+      val setter = methods.find{it.selector.startsWith("set${varName}") && it.parameters.size == 1}
```

Comment:
So just to be clear, setters won't always be defined (resulting in "val" properties in Kotlin), so please make sure this code is ready for that.  Also, consider comparing the getter's return type and setters parameter type to make sure they match or are at least related (one a subtype of the other).  Finally, if the getter is static, the setter should probably also be.

Reply:
I think checking the types match makes sense.

I can't find any good formal semantics on how Kotlin makes this conversion.  The best I've found is here: https://kotlinlang.org/docs/reference/java-interop.html#getters-and-setters

Is there a more formal reference?

Reply:
From the internal code, cinterop will create a val if the setter is not defined so it handles that at least.

Implemented the checks.

Couldn't find anything more than that page for reference.